x/concentrated-liquidity/types/keyspace.md

```diff
@@ -0,0 +1,107 @@
+# CL keyspace
+
+This document defines the key space format used in CL, and guarantees that are expected to hold.
+It was created to more easily reason about key space bugs that we were facing.
+
+If || is seen outside of a `backtick` quotation, it means raw appending.
+
+We use `string encoding` of an integer to mean the variable length, base10 encoding of an integer into a string.
+
+Also all of this really shows we vastly overpay for complexity in the statemachine, and that collections work in the SDK can really simplify this code.
+
+We break this up into parts, the multi-component keys, and the single component keys that don't really have concern.
+
+TODO: How do we train AI tools to do more of this for us, instead if it being manual
+- Maybe not worth doing since collections / ORM should delete this code for us.
+
+## multi-component keys
+
+## 0x01 - Pool Tick storage
+
+If a key exists in state, that begins with `0x01`, it is expected that it is of the form:
+`0x01` || `8 byte big endian encoding of pool ID` || `9 byte signed tick encoding`
+
+We are expected to be able to iterate over all ticks in a pool, from most negative to most positive.
+
+## 0x02 - Indexed position ID storage
+
+If a key exists in state, that begins with `0x02`, it is expected that it is of the form:
+`0x02/` || `Hexadecimal encoding of an address` || `/` || `string encoding of pool ID` || `/` || `string encoding of position ID`
```

Comment:
I don't think it's hex but I will double check

Reply:
wrote a test to show its hex